test/e2e/specs/customer_admin.go

```diff
@@ -233,15 +236,26 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
 	It("should sync AAD admin group", func() {
 		syncSecret, err := sanity.Checker.Client.Admin.CoreV1.Secrets("openshift-infra").Get("aad-group-sync-config", metav1.GetOptions{})
 		Expect(err).ToNot(HaveOccurred())
-		id := "44e69b4e-2e70-42df-bb97-3a890730d7b0"
-		testUser := "testuserdisabled"
-		gid := syncSecret.Data["customerAdminGroupId"]
-		if strings.EqualFold(string(gid), id) {
-			// test the users
-			oca, err := sanity.Checker.Client.Admin.UserV1.Groups().Get("osa-customer-admins", metav1.GetOptions{})
+
+		var provider api.AADIdentityProvider
+		err = yaml.Unmarshal(syncSecret.Data["aad-group-sync.yaml"], &provider)
+		Expect(err).ToNot(HaveOccurred())
+
+		if provider.CustomerAdminGroupID != nil &&
+			*provider.CustomerAdminGroupID == "44e69b4e-2e70-42df-bb97-3a890730d7b0" {
```

Comment:
shouldn't we drop this test for the gid? it makes it hard to reconfigure the test without breaking the test. Also shouldn't the test fail if provider.CustomerAdminGroupID == nil? otherwise we silently skip the test if groupId is not set. and if running locally you don't know that a test is not running.

Reply:
it's tricky.  The e2e test doesn't have permissions to query Azure to read the group memberships, so it's hard coded.  I don't want the test to fail on clusters that are not configured with that group id.  The downside is that it's still easy for the test to skip and for us not to realise that the functionality is broken :-/